java/fury-core/src/main/java/org/apache/fury/serializer/AbstractObjectSerializer.java

```diff
@@ -398,27 +402,55 @@ private static FinalTypeField buildFinalTypeField(Fury fury, Descriptor d) {
         d.getDeclaringClass() + "." + d.getName(),
         // `d.getField()` will be null when peer class doesn't have this field.
         d.getField() != null ? FieldAccessor.createAccessor(d.getField()) : null,
-        fury);
+        fury,
+        d.getField() != null ? d.getField().getAnnotation(FuryField.class) : null);
   }
 
   private static GenericTypeField buildContainerField(Fury fury, Descriptor d) {
     return new GenericTypeField(
         d.getTypeRef(),
         d.getDeclaringClass() + "." + d.getName(),
         d.getField() != null ? FieldAccessor.createAccessor(d.getField()) : null,
-        fury);
+        fury,
+        d.getField() != null ? d.getField().getAnnotation(FuryField.class) : null);
```

Comment:
How about parse Annotation in Descriptor? And pass descriptor into `GenericTypeField`

Reply:
We could pass Descriptor into GenericTypeField instead of pass `FieldAccessor`. Then creating `FieldAccessor` in `GenericTypeField`

Reply:
`org.apache.fury.builder.BaseObjectCodecBuilder` also needs to read this annotation data